patches/api/0369-Added-and-implemented-PlayerLootDropEvent.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: NeonOverflow <nneonoverflow@protonmail.com>
+Date: Thu, 13 Jan 2022 22:58:50 -0500
+Subject: [PATCH] Added and implemented PlayerLootDropEvent.
+
+As a result of how the PlayerDeathEvent was implemented, when a player dies their items are simply dropped
+through the "world.dropItem()" method. This makes it very annoying for plugin developers to ultimately
+track down the Item entities dropped when a player dies. This patch implements a "PlayerLootDropEvent",
+making it easy for developers to track down the Item entities dropped when a player dies. Furthermore,
+Item entities dropped on player death are marked as being thrown by the player that died.
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerLootDropEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerLootDropEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..739ef6e6a06df07f4031da8cca3255948ab63a8c
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerLootDropEvent.java
+@@ -0,0 +1,56 @@
++package io.papermc.paper.event.player;
++
++import java.util.Collections;
++import java.util.List;
++
++import org.bukkit.entity.Item;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Is called when a {@link org.bukkit.entity.Player} drops their {@link org.bukkit.entity.Item}s on death.
```

Comment:
Unlike most events this actually happens **after** all the item entities are created, it might be good to mention that here, that while it all happens in the same tick, and right after each other, technically the event is being fired after the item entities exist in the world.